### ISSUE TITLE
Publish a new release with client-go v0.27 support

### ISSUE THREAD

**Issue by howardjohn** [kubernetes-sigs/controller-runtime#2295]
v0.27 is supported since https://github.com/kubernetes-sigs/controller-runtime/pull/2223 is released. This is usable today fi you don't mind pinning to a random commit instead of a tag.

However, the HEAD is reporting as `v0.13.1-0.20230315234915-a26de2d610c3` due to the branching process used. This makes Go treat this as ~v0.13.2. If you try to import a library using v0.14.x, it sees this as older and reverts. This requires a `replace`, which causes all sorts of problems.

tl;dr, you cannot use client-go v0.27 and controller-runtime AND import something that uses controller-runtime v0.14

Cutting a release of v0.15 would fix this and make it so the broader ecosystem can adopt client-go v0.27

**Comment by vincepri** [comment]
This is coming soon, we're waiting for #2297

**Comment by vincepri** [comment]
/close

Released [v0.15.0-alpha.0](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0-alpha.0) tag for folks to consume if you all need this now. The v0.15.0 release should be ready in a couple of weeks, following Kubernetes v1.27.2.

**Comment by sebrandon1** [comment]
Agreed.  Also waiting on this for multiple projects.
